src/db/options.rs

```diff
@@ -312,3 +312,34 @@ pub struct ChangeStreamPreAndPostImages {
     /// If `true`, change streams will be able to include pre- and post-images.
     pub enabled: bool,
 }
+
+/// Determines the lifetime of a cursor.
+#[derive(Clone, Debug, Deserialize)]
+#[non_exhaustive]
+pub enum TimeoutMode {
+    /// Times out after one Iteration.
+    Iteration,
+    /// Times out after the end of the Cursor Lifetime.
+    CursorLifetime,
+}
+
+/// Specifies the options to a
+/// [`Database::RunCursorCommand`](../struct.Database.html#method.run_cursor_command) operation.
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[serde(rename_all = "camelCase")]
+#[builder(field_defaults(default, setter(into)))]
+#[non_exhaustive]
+pub struct RunCursorCommandOptions {
```

Comment:
I just noticed that [the spec](https://github.com/mongodb/specifications/blob/master/source/run-command/run-command.rst#runcursorcommand) has `RunCursorCommandOptions` extending `RunCommandOptions`.  We don't actually have `RunCommandOptions` in the Rust driver (this was a mistake, one of the things we want to fix in 3.0), so we'll need to fold [those options](https://github.com/mongodb/specifications/blob/master/source/run-command/run-command.rst#runcommand) from the spec into `RunCursorCommandOptions`:
* `readPreference`: this can just be directly added to `RunCursorCommandOptions`, it'll be used to supply the `SelectionCriteria` for `RunCommand::new`
* `session`: because of Rust lifetime restrictions, this can't be directly included in the options; instead, we have to kind of awkwardly duplicate most methods to provide one that doesn't accept a session and one that does.  See `run_command` / `run_command_with_session` for an example.
* `timeoutMS`: this is for client-side operation timeouts (CSOT), which the Rust driver doesn't support, so you can just skip it.